### PR TITLE
Set correct minimal versions in default conf

### PR DESCRIPTION
**Pull request by RonEld**
Set `MBEDTLS_SSL_MIN_MAJOR_VERSION` and `MBEDTLS_SSL_MIN_MINOR_VERSION`
instead of `MBEDTLS_SSL_MAJOR_VERSION_3` and `MBEDTLS_SSL_MINOR_VERSION_1`

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
Internal ref: IOTSSL-1387

**Comment by gilles-peskine-arm** [comment]
Needs backport to 2.1 → PR #940

**Comment by andresag01** [comment]
Also, please note that there are other places where we are setting the default minor/major to a hard value even though that particular version might have been disabled at compile-time. For example: https://github.com/ARMmbed/mbedtls/pull/939/files#diff-995204bbb15401337db9b64bc314e1f6L7280. That might be OK, but I still thought I would point it out

**Comment by andresag01** [comment]
This PR addresses https://github.com/ARMmbed/mbedtls/issues/664

**Comment by mpg** [comment]
I disagree with this change. While I agree that the previous behaviour was wrong, I think what we really want is to set min_version to the highest of SSL_MIN_VERSION and TLS 1.0, ie, never set it to SSLv3 by default.

Obsolete things that are no longer secure should need to be explicitly enabled both at compile-time and at runtime. An example where this matters is people using us as a shared library: they might use a very full build that includes obsolete things in case someone needs them, but that shouldn't make those things enabled by default for everybody.

**Comment by RonEld** [comment]
@mpg I understand your concern and agree with you. I hope this new update will please you.
I am not sure about the name of the new definition, and the location - should I have put it in `ssl_internal.h`?

**Comment by mpg** [comment]
Looks good to me! But check-names .sh thinks that macros defined in .c files should start with the module name (without the `MBEDTLS_` prefix). So you can either drop this prefix or move the definition to `ssl_internal.h`, I think both are OK.

**Comment by RonEld** [comment]
@mpg Thanks for the reminder. 
I moved the definitions to *`ssl_internal.h`*

**Comment by RonEld** [comment]
@andresag01 about your comment about other locations, I think you are referring to nist suiteB, right?
This is setting TLS 1.2 as default, and I believe this is according to the test suiteB

File: library/ssl_tls.c [hunk 1]
```diff
@@ -7304,8 +7304,8 @@ int mbedtls_ssl_config_defaults( mbedtls_ssl_config *conf,
          * Default
          */
         default:
-            conf->min_major_ver = MBEDTLS_SSL_MAJOR_VERSION_3;
-            conf->min_minor_ver = MBEDTLS_SSL_MINOR_VERSION_1; /* TLS 1.0 */
+            conf->min_major_ver = ( MBEDTLS_SSL_MIN_MAJOR_VERSION > MBEDTLS_SSL_MIN_VALID_MAJOR_VERSION ) ? MBEDTLS_SSL_MIN_MAJOR_VERSION : MBEDTLS_SSL_MIN_VALID_MAJOR_VERSION;
+            conf->min_minor_ver = ( MBEDTLS_SSL_MIN_MINOR_VERSION > MBEDTLS_SSL_MIN_VALID_MINOR_VERSION ) ? MBEDTLS_SSL_MIN_MINOR_VERSION : MBEDTLS_SSL_MIN_VALID_MINOR_VERSION;
```
Comment (andresag01) [inline library/ssl_tls.c]: Style: Could you please wrap this lines at 80 characters?

Comment (RonEld) [inline library/ssl_tls.c]: done

**Comment by RonEld** [comment]
@andresag01 I updated according to your style coment

**Comment by andresag01** [comment]
@RonEld: Thanks for making the style change. I have approved the PR once again.

**Comment by RonEld** [comment]
@gilles-peskine-arm since you did the backport on #940 , could you update that PR according to the comments here?

**Comment by mpg** [comment]
Note: since IOTSSL-1387 has been closed as a duplicate, the internal ref for this is not IOTSSL-1072.

Note: I think this issue is not present is 1.3 so the fix doesn't need to be backported to that branch (I guess you guys are already aware of it since there is a backport for 2.1 only, but I didn't see it stated explicitly, so here it is for the record).

**Comment by gilles-peskine-arm** [comment]
retest

**Comment by gilles-peskine-arm** [comment]
Superseded by #1296 (same patch, rebased to current `development`)

**Comment by gilles-peskine-arm** [comment]
Superseded by #1296 (same patch, rebased to current `development`)

**Pull request closed without merging**